app/controllers/requests.rb

```diff
@@ -0,0 +1,89 @@
+class MakersBnb < Sinatra::Base
+
+  get '/requests' do
+    if current_user
+      @users_requests = Request.all(user_id: current_user.id)
+      @received_requests = Request.all(owner_id: current_user.id)
+      @request_display = @users_requests.map do |x|
+        Space.get(x.space_id)
+      end
+      erb :'requests/index'
+    else
+      redirect to '/sessions/new'
+    end
+  end
+
+  post '/requests/new' do
+
+    requested_date = Date.parse(params[:date_requested]) rescue nil
+
+    unless requested_date
+      flash[:notice] = 'Please enter a valid date'
+      redirect '/spaces'
+    end
+
+    unless session[:available_dates].include?(Date.parse(params[:date_requested]).strftime)
+      flash[:notice] = 'Space not available on the requested date'
+      redirect '/spaces'
+    end
+
+    if current_user
+      unless current_user.id != Space.get(params[:space_id]).user_id
+        flash[:notice] = 'This is your own space'
+        redirect '/spaces'
+      end
+      date = Date.parse(params[:date_requested])
+      space = Space.get(params[:space_id])
+      request = Request.create(user_id: current_user.id, space_id: params[:space_id], date_requested: date, owner_id: space.user_id)
+      flash.keep[:notice] = 'Booking requested'
+      redirect '/users/account'
+    else
+      flash[:notice] = 'Please log in to request a space'
+      redirect '/spaces'
+    end
+  end
```

Comment:
Note that you'll have to define a bunch of methods on the model to make that work.